tf2onnx/graph.py

```diff
@@ -1195,13 +1269,20 @@ def insert_new_node_on_output(self, op_type, output_name, name, domain=None, **k
         new_node = self.make_node(op_type, [output_name], attr=kwargs, outputs=[new_output], name=name, domain=domain)
 
         to_replace = [n for n in self.get_nodes() if n != new_node]
-        self.replace_all_inputs(to_replace, output_name, new_output)
+        self.replace_all_inputs(to_replace, output_name, new_output, keep_ops=True)
         return new_node
 
     def find_output_consumers(self, output_name):
         """Find all nodes consuming a given output."""
+        if output_name in self._input_to_node_name:
```

Comment:
Using _input_to_node_name is relevant to find all nodes ingesting output_name in this graph. But that does not include subgraphs.